app/code/Magento/CompareListGraphQl/Model/Service/AddToCompareListService.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CompareListGraphQl\Model\Service;
+
+use Magento\Catalog\Model\Product;
+use Magento\Catalog\Model\Product\Compare\Item;
+use Magento\Catalog\Model\Product\Compare\ItemFactory;
+use Magento\Catalog\Model\ProductRepository;
+use Magento\Framework\Exception\NoSuchEntityException;
+
+/**
+ * Service add product to compare list
+ */
+class AddToCompareListService
```

Comment:
You may use `AddToCompareList` as a name, without suffixing with the "...Service". Also, I would recommend renaming the entry point (public function) to `execute` in this way we will have more or less consistent naming for the method invocation in other classes

Reply:
**Done**